src/library.js

```diff
@@ -2304,6 +2304,7 @@ LibraryManager.library = {
         {{{ makeSetValue('sa', C_STRUCTS.sockaddr_in.sin_family, 'family', 'i16') }}};
         {{{ makeSetValue('sa', C_STRUCTS.sockaddr_in.sin_addr.s_addr, 'addr', 'i32') }}};
         {{{ makeSetValue('sa', C_STRUCTS.sockaddr_in.sin_port, '_htons(port)', 'i16') }}};
+        _memset(sa + {{{ C_STRUCTS.sockaddr_in.sin_zero }}}, 0, {{{ C_STRUCTS.sockaddr_in.__size__ - C_STRUCTS.sockaddr_in.sin_zero }}});
```

Comment:
We use makeSetValue with i64 to avoid the `_memset` dependency?   

We can add a compile time assert to check its valid.   `{{{ assert(C_STRUCTS.sockaddr_in.__size__ - C_STRUCTS.sockaddr_in.sin_zero == 8) }}}`

Reply:
Sure, I will do that in a bit.

Just for my education, is there any issue with creating the `_memset` dependency? Other places uses it too, which makes me wonder why that would be an issue :)

Reply:
Its just that it adds code size I think.

We should probably audit the various places we use it in JavaScript and try to be consistent.  There are several places IIRC where we use a loop in JS instead.  For example: https://github.com/emscripten-core/emscripten/blob/585c5e6990c6bb54c437ee18dc658bc4f958c36b/src/library_pthread.js#L757

@kripken @juj do you two think?  When should we use _memset and when should we not?  Is it just a code size consideration?  Perhaps we could make a `{{{ doMemset }}}` replacement that could do `if (memset_exported) return "_memset" else return "memsetInJs` ?

Reply:
Yeah, adding code size over this would be a bit unneeded :)

Changed it to `makeSetValue` with an `assert` guarding it; works as well as memset, and makes the PR looks nice and tiny :D Tnx!

Reply:
A `{{{ doMemSet(ptr, align, size) }}}` function does sound fine to me - it could compile-time unroll some small values in -O1 and higher, cutoff value perhaps depending on the -On level, and in -Os values beyond the cutoff could call to a JS function to do the job, if memset was not exported.

Btw, is there a way to have a generic `#if is_exported('memset')` functionality in JS code? Would prefer that over special-casing `memset` symbol specifically for this. That might allow creating quite tight code size aware codegen in user JS libraries.

Reply:
Yes you can say `#if '_malloc' in IMPLEMENTED_FUNCTIONS`

Reply:
A `doMemSet` sounds good to me too.

@juj 

> Btw, is there a way to have a generic #if is_exported('memset')

Yes, you can do stuff like
```
#if hasExportedFunction('_main')
```